trl/trainer/ppo_trainer.py

```diff
@@ -821,8 +854,12 @@ def step(
         train_stats["policy/advantages"] = torch.nan_to_num(train_stats["policy/advantages"], WANDB_PADDING)
         train_stats["policy/ratio"] = torch.flatten(train_stats["policy/ratio"]).unsqueeze(0)
 
+        total_scores = torch.tensor([score[score != padding_value].sum() for score in padded_scores]).to(
+            self.current_device
+        )
+
```

Comment:
We should not only sum it up — the discount factor gamma also comes in. The total_scores should be the discounted episodic return. See https://spinningup.openai.com/en/latest/spinningup/rl_intro.html

<img width="740" alt="image" src="https://github.com/huggingface/trl/assets/5555347/977558aa-e0cd-4579-97db-ac8245bb4620">

Reply:
@vwxyzjn ah yes good point, I've added that calculation